### PR TITLE
Added ESP32 telemetry option

### PR DESCRIPTION
**Pull request by seeul8er**

### CODE REVIEW

File: common/source/docs/common-esp32-telemetry.rst [hunk 1]
```diff
@@ -0,0 +1,210 @@
+.. _common-esp32-telemetry:
+
+======================
+ESP32 WiFi telemetry
+======================
+
+The ESP32 are readily available Wi-Fi modules with full TCP/IP stack and
```
Comment (Hwurzburg) [inline common/source/docs/common-esp32-telemetry.rst:7]: advantages vs ESP-x 8226 for this application? which requires no programming, just wiring to autopilot

File: common/source/docs/common-esp32-telemetry.rst [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+   -  Stable 3.3V or 5V power supply to the ESP32 (depending on the
+      available inputs of your DevKit)
+
+-  Set the flight controller port to the desired protocol.
+
+Some manufacturers of ESP32 DevKits have wrong labels for the
+pins on their products. Make sure that the PINs on your board are
+labeled correctly if you encounter issues.
+
+Configuring GCS like QGroundControl
```
Comment (Hwurzburg) [inline common/source/docs/common-esp32-telemetry.rst:128]:
```suggestion
Connecting to GCS
```

File: common/source/docs/common-esp32-telemetry.rst [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+===================================
```
Comment (Hwurzburg) [inline common/source/docs/common-esp32-telemetry.rst:129]: adjust length if title is changed

File: common/source/docs/common-esp32-telemetry.rst [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+The following connection options are available:
+
+-  UDP unicast on port ``14550`` to all connected devices.
+-  TCP on port ``5760``
+
+DroneBridge for ESP32 will automatically forward all data to all
+connected WiFi devices via UDP to port 14550. QGroundControl should
```
Comment (Hwurzburg) [inline common/source/docs/common-esp32-telemetry.rst:137]:
```suggestion
connected WiFi devices via UDP to port 14550. QGroundControl or Mission Planner should
```

File: common/source/docs/common-esp32-telemetry.rst [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+   -  GND to GND
+   -  Stable 3.3V or 5V power supply to the ESP32 (depending on the
+      available inputs of your DevKit)
+
+-  Set the flight controller port to the desired protocol.
+
+Some manufacturers of ESP32 DevKits have wrong labels for the
+pins on their products. Make sure that the PINs on your board are
+labeled correctly if you encounter issues.
+
```
Comment (Hwurzburg) [inline common/source/docs/common-esp32-telemetry.rst:127]: Need autopilot configuration section

**Review by Hwurzburg — CHANGES_REQUESTED** [review]

**Comment by Hwurzburg** [comment]
will also need to add to telemetry landing page TOC

**Comment by seeul8er** [comment]
Thank you for your feedback. I'll get behind it

File: common/source/docs/common-esp32-telemetry.rst (hunk 1 above)
Comment (seeul8er) [reply common/source/docs/common-esp32-telemetry.rst:7]: The ESP32 is faster and does consume less power when WiFi encryption is enable. Most of the benefits the ESP32 has over the ESP8226 do not matter when is comes to this application. It all depends what kind of module you got lying around and is ready to use. The ESP8226 does not come pre-flashed - they are general purpose. I think there is a vendor that sells the 8266 module already pre-flashed.
The ESP32 has a lot of power left to be used for other applications - like doing additional sensors etc.

File: common/source/docs/common-esp32-telemetry.rst (hunk 5 above)
Comment (seeul8er) [reply common/source/docs/common-esp32-telemetry.rst:132]: updated

File: common/source/docs/common-esp32-telemetry.rst (hunk 3 above)
Comment (seeul8er) [reply common/source/docs/common-esp32-telemetry.rst]: done

File: common/source/docs/common-esp32-telemetry.rst (hunk 5 above)
Comment (seeul8er) [reply common/source/docs/common-esp32-telemetry.rst:132]: You mean Mission Planner config to work with the ESP?

**Comment by seeul8er** [comment]
Added to telemetry landing page

File: common/source/docs/common-esp32-telemetry.rst (hunk 5 above)
Comment (Hwurzburg) [reply common/source/docs/common-esp32-telemetry.rst:132]: no, for the autopilot UART port to work with it...correct protocol and baud rate must be setup on desired port

Comment (seeul8er) [reply common/source/docs/common-esp32-telemetry.rst:132]: Added

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
since I dont have access to make changes...can you squash this to a single commit....I have a few nits but will fix them after I merge...

File: common/source/docs/common-esp32-telemetry.rst [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+   :alt: Example wiring of flight controller to ESP32
+
+
+ArduPilot configuration
+=======================
+
+Configure the UART of the flight controller that is wired to the ESP32. The default configuration of DroneBridge is:
+
+-  Protocol: MAVLink (v1 or v2)
+-  Baudrate: 115200 mbps
```
[the remark this answers is not in the document]
Comment (seeul8er) [reply common/source/docs/common-esp32-telemetry.rst]: fixed

**Comment by seeul8er** [comment]
Made a single commit

**Comment by Hwurzburg** [comment]
thanks...merging

**Pull request merged**